core/jvm-native/src/main/scala/cats/effect/unsafe/PollingSystem.scala

```diff
@@ -29,19 +46,49 @@ abstract class PollingSystem {
    */
   type Poller <: AnyRef
 
+  /**
+   * Provides the functionality to close the polling system.
+   */
   def close(): Unit
 
+  /**
+   * Creates a new instance of the user-facing interface.
+   *
+   * @param register
+   *   callback that takes a function from `Poller` to `Unit` The function is used to register a
+   *   new poller.
```

Comment:
> The function is used to register a new poller

This is not correct, it's used to obtain an existing poller, not to register a new one. I can see why it's confusing, because it's named "register". I think it was originally Daniel's idea in one of the older prototypes, maybe he can explain it, or we should consider a different name?

```suggestion
   * @param register
   *   callback to obtain a thread-local `Poller`.
```

Reply:
I had a conversation with Daniel and he clarified me how it works. We all agrees that the name is confusing and we thought about renaming `register` to `access`. What do you think ? :)

Reply:
Thanks for discussing that! I like `access` 👍